src/reconcile/rec_write.c

```diff
@@ -2426,8 +2431,12 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
      * visible when reconciling this page, copy them into the database's history store. This can
      * fail, so try before clearing the page's previous reconciliation state.
      */
-    if (F_ISSET(r, WT_REC_HS))
-        WT_RET(__rec_hs_wrapup(session, r));
+    if (F_ISSET(r, WT_REC_HS)) {
+        session->reconcile_timeline.hs_wrapup_start = __wt_clock(session);
+        ret = __rec_hs_wrapup(session, r);
+        session->reconcile_timeline.hs_wrapup_finish = __wt_clock(session);
+        WT_RET(ret);
+    }
```

Comment:
There exists a stat indicating how much time a maximum reconciliation call is spent `rec_maximum_seconds`. Can you add three more stats with the above-calculated information along with the warning?

Reply:
@bvpvamsikrishna @agorrod I think this is a good idea. We can add three more stats max_evict_seconds, max_disk_build_seconds, max_hs_wrapup_seconds. What's your opinion?

Reply:
sounds good to me